### PR TITLE
Upgrade to Xcode 12

### PR DESCRIPTION
**Pull request by clausjoergensen**
- Removed Carthage as it's not compatible with Xcode 12
- Updated to the latest Quick&Nimble for Xcode 12 support.
- Ran the linter (for what appears to be the first time in a while)
- Fixed swiftlint errors.
- Fixed various compilation errors in the tests.
- Disabled Jazzy's check for undocumented methods, as this clearly hadn't been checked for a very long time.
- Updated CI scripts to reflect the above changes.

### CODE REVIEW

**Comment by clausjoergensen** [comment]
These tests seems to be unstable

> Failing tests:
> 	SchibstedAccountTests:
> 		IdentityManagerTests.Login_with_password__Should_pass_in_a_credential()
> 		TaskManagerTests.Adding_a_task__should_not_refresh_if_already_in_progress()
> 		TaskManagerTests.Adding_a_task__should_not_refresh_if_already_in_progress()

**Review by zamzterz — APPROVED** [review]
Much appreciated for a long overdue cleanup! 🙇 

Those flaky tests pass on my local machine, so I'd opt for keeping them for now (it's not that hard verifying it's those same requests failing in future Travis builds).

File: Tests/URLSessionTests.swift [hunk 1]
```diff
@@ -261,10 +261,6 @@ class URLSessionTests: QuickSpec {
                 SDKConfiguration.shared.refreshRetryCount = 2
                 doDataTask(session, url: URL(string: wantedUrl)!) { _, _, error in
                     expect(error).to(matchError(ClientError.userRefreshFailed(kDummyError)))
-                    guard let clientError = error as? ClientError, case let ClientError.userRefreshFailed(error) = clientError else {
```
[the remark this answers is not in the document]
Comment (clausjoergensen) [reply Tests/URLSessionTests.swift:264]: Caused the test to always fail, and it didn't seem to actually contribute anything to the test itself. But maybe we should revert it and just let it fail on Travis for a bit

[the remark this answers is not in the document]
Comment (zamzterz) [reply Tests/URLSessionTests.swift:264]: It doesn't pass on local either, so let's keep it removed 🙂

**Pull request merged**